precompiles/erc20factory/IERC20Factory.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.17;
+
+/**
+ * @dev The ERC20 Factory contract's address.
+ */
+address constant ERC20_FACTORY_PRECOMPILE_ADDRESS = 0x0000000000000000000000000000000000000900;
+
+/**
+ * @dev The ERC20 Factory contract's instance.
+ */
+IERC20Factory constant ERC20_FACTORY_CONTRACT = IERC20Factory(ERC20_FACTORY_PRECOMPILE_ADDRESS);
+
+interface IERC20Factory {
+    /**
+     * @dev Defines a method for creating an ERC20 token.
+     * @param tokenPairType Token Pair type
+     * @param salt Salt used for deployment
+     * @param name The name of the token.
+     * @param symbol The symbol of the token.
+     * @param decimals the decimals of the token.
+     * @return tokenAddress The ERC20 token address.
+     */
+    function create(
```

Comment:
I think this should include an initial (validated) mint amount that gets sent to the creator of the denom.